wcc-contentful/lib/wcc/contentful/simple_client/http_adapter.rb

```diff
@@ -3,22 +3,36 @@
 gem 'http'
 require 'http'
 
-class HttpAdapter
-  def call(url, query, headers = {}, proxy = {})
-    if proxy[:host]
-      HTTP[headers].via(proxy[:host], proxy[:port], proxy[:username], proxy[:password])
-        .get(url, params: query)
-    else
-      HTTP[headers].get(url, params: query)
-    end
+class WCC::Contentful::SimpleClient::HttpAdapter
+  def get(url, params = {}, headers = {})
+    req = OpenStruct.new(params: params, headers: headers)
+
+    yield req if block_given?
+
+    Response.new(
+      HTTP[req.headers].get(url, params: req.params)
+    )
   end
 
   def post(url, body, headers = {}, proxy = {})
-    if proxy[:host]
-      HTTP[headers].via(proxy[:host], proxy[:port], proxy[:username], proxy[:password])
-        .post(url, json: body)
-    else
-      HTTP[headers].post(url, json: body)
-    end
+    Response.new(
+      if proxy[:host]
+        HTTP[headers].via(proxy[:host], proxy[:port], proxy[:username], proxy[:password])
+          .post(url, json: body)
+      else
+        HTTP[headers].post(url, json: body)
+      end
+    )
   end
+
+  Response =
+    Struct.new(:raw) do
+      extend Forwardable
+
+      def_delegators :raw, :body, :to_s, :status, :headers
```

Comment:
`status` intended to be here since it's defined in class?

Reply:
Good catch